Modix.Services/Moderation/ModerationInvitePurgingHandler.cs

```diff
@@ -1,64 +1,56 @@
 using System;
 using System.Linq;
 using System.Text.RegularExpressions;
+using System.Threading;
 using System.Threading.Tasks;
 
 using Discord;
 using Discord.WebSocket;
-
+using MediatR;
 using Modix.Data.Models.Core;
 using Modix.Services.Core;
+using Modix.Services.Messages.Discord;
 using Serilog;
 
 namespace Modix.Services.Moderation
 {
     /// <summary>
-    /// Implements a behavior that automatically deletes invite links posted by select users.
+    /// Implements a handler that automatically deletes invite links posted by select users.
     /// </summary>
-    public class ModerationInvitePurgingBehavior : BehaviorBase
+    public class ModerationInvitePurgingHandler :
+        INotificationHandler<ChatMessageReceived>,
+        INotificationHandler<ChatMessageUpdated>
     {
-        // TODO: Abstract DiscordSocketClient to IDiscordSocketClient, or something, to make this testable
+        private readonly IDesignatedChannelService _designatedChannelService;
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IModerationService _moderationService;
+        private readonly IUser _botUser;
+
         /// <summary>
-        /// Constructs a new <see cref="ModerationInvitePurgingBehavior"/> object, with the given injected dependencies.
+        /// Constructs a new <see cref="ModerationInvitePurgingHandler"/> object, with the given injected dependencies.
         /// See <see cref="BehaviorBase"/> for more details.
         /// </summary>
-        /// <param name="discordClient">The value to use for <see cref="DiscordClient"/>.</param>
-        /// <param name="serviceProvider">See <see cref="BehaviorBase"/>.</param>
-        public ModerationInvitePurgingBehavior(DiscordSocketClient discordClient, IServiceProvider serviceProvider)
-            : base(serviceProvider)
-        {
-            DiscordClient = discordClient;
-        }
-
-        /// <inheritdoc />
-        internal protected override Task OnStartingAsync()
-        {
-            DiscordClient.MessageReceived += OnDiscordClientMessageReceived;
-            DiscordClient.MessageUpdated += OnDiscordClientMessageUpdated;
-
-            return Task.CompletedTask;
-        }
-
-        /// <inheritdoc />
-        internal protected override Task OnStoppedAsync()
+        /// <param name="designatedChannelService"></param>
+        /// <param name="authorizationService"></param>
+        /// <param name="moderationService"></param>
+        /// <param name="botUser"></param>
+        public ModerationInvitePurgingHandler(
+            IDesignatedChannelService designatedChannelService,
+            IAuthorizationService authorizationService,
+            IModerationService moderationService,
+            IUser botUser)
```

Comment:
Shouldn't this be `ISelfUser`? That's how the bot user is registered in `ServiceCollectionExtensions.AddModix`.